onadata/libs/utils/osm.py

```diff
@@ -166,45 +168,51 @@ def save_osm_data(instance_id):
         }
 
         for osm in osm_attachments:
-            osm_xml = osm.media_file.read()
-            filename = None
-            field_name = None
-            for k, v in osm_filenames.items():
-                if osm.filename.startswith(v.replace('.osm', '')):
-                    filename = v
-                    field_name = k
-                    break
-
-            if field_name is None:
-                continue
-            filename = osm.filename if filename is None else filename
-            osm_list = parse_osm(osm_xml, include_osm_id=True)
-            for osmd in osm_list:
-                try:
-                    with transaction.atomic():
-                        osm_data = OsmData(
-                            instance=instance,
-                            xml=osm_xml,
-                            osm_id=osmd['osm_id'],
-                            osm_type=osmd['osm_type'],
-                            tags=osmd['tags'],
-                            geom=GeometryCollection(osmd['geom']),
-                            filename=filename,
-                            field_name=field_name)
-                        osm_data.save()
-                except IntegrityError:
-                    with transaction.atomic():
-                        osm_data = OsmData.objects.exclude(
-                            xml=osm_xml).filter(
-                            instance=instance, field_name=field_name).first()
-                        if osm_data:
-                            osm_data.xml = osm_xml
-                            osm_data.osm_id = osmd['osm_id']
-                            osm_data.osm_type = osmd['osm_type']
-                            osm_data.tags = osmd['tags']
-                            osm_data.geom = GeometryCollection(osmd['geom'])
-                            osm_data.filename = filename
+            try:
+                osm_xml = osm.media_file.read()
+            except IOError as e:
+                raise Http404(str(e))
```

Comment:
Why Http404?

Reply:
Http404 will not make sense when this is run by celery as an async task. I will recommend failing silently, and simply skip to the next attachment on the list.